server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -142,6 +144,18 @@
 @LuceneTestCase.SuppressFileSystems("ExtrasFS")
 public class TranslogTests extends ESTestCase {
 
+    public static final DiskIoBufferPool RANDOMIZING_IO_BUFFERS = new DiskIoBufferPool() {
+        @Override
+        public ByteBuffer maybeGetDirectIOBuffer() {
+            final String currentThreadName = Thread.currentThread().getName();
+            try {
+                Thread.currentThread().setName(randomBoolean() ? "[" + ThreadPool.Names.WRITE + "] thread" : "not-a-write-thread");
+                return super.maybeGetDirectIOBuffer();
```

Comment:
Can we assert that we get null when the name is `not-a-write-thread` and a direct buffer when it is `[WRITE]`? Just to guard against the check changing in `maybeGetDirectIOBuffer`, nulling out the effect of this test (as well as to check that the test works now based on the name check).

Reply:
++ done, had to expose the thread-local for this because we run quite a few calls to this on test threads but that's ok IMO :)